pyproject.toml

```diff
@@ -1,3 +1,136 @@
 [build-system]
-requires = ["setuptools >= 40.6.0", "wheel"]
+requires = ["setuptools", "wheel"]
 build-backend = "setuptools.build_meta"
+
+[project]
+name = "smarthomeconnect"
+version = "0.9.0"
+description = "The Smart Home Connect home automation framework based on AsyncIO"
+license = {file = "LICENSE"}
+readme = "README.md"
+authors = [{name = "Michael Thies", email = "mail@mhthies.de"}]
+requires-python = ">=3.7"
+dependencies = [
+    "aiohttp>=3.6,<4",
+    "jinja2>=2.11,<4",
+    "MarkupSafe>=1.1,<3",
+]
+
+[tool.setuptools]
+packages = ["shc"]
+
+[project.optional-dependencies]
+mysql = ["aiomysql>=0.0.21,<=0.2.0"]
+knx = ["knxdclient>=0.4.0,<2"]
+dmx = ["pyserial-asyncio>=0.3,<0.7"]
+midi = ["mido>=1.2.9,<2", "python-rtmidi>=1.4.6,<2"]
+mqtt = ["aiomqtt>=2.0.0,<3"]
+pulse = ["pulsectl_asyncio>=1.0.0,<2"]
+telegram = ["aiogram>=3.0,<4"]
+file_persistence = ["aiofile>=3.7.4,<4"]
+dev = ["ruff~=0.8"]
+test = [
+    "coverage~=7.6",
+    "mypy~=1.13",
+    "selenium>=3.141,<5",
+    "types-PyMySQL",
+    "types-MarkupSafe>=1.1,<3",
+    "types-Jinja2>=2.11,<3",
+]
+doc = [
+    "sphinx~=5.0",
+    "sphinx-rtd-theme~=2.0",
+    "sphinx-toolbox~=3.5",
+    "sphinxcontrib-httpdomain~=1.8",
+]
+
+[tool.ruff]
+line-length = 120
+output-format = "concise"
+exclude = [
+    ".cache",
+    ".git",
+    "dist",
+    "docs",
+    "htmlcov",
+    "node_modules",
+    "venv",
+    "venv*",
+    ".mypy_cache",
+    "test/assets",
+]
+
+[tool.ruff.lint]
+# Enable checks for Pycodestyle, Pyflakes, Bugbear, Docstrings and Import sorting
+select = ["E", "F", "B", "D", "I"]
+ignore = [
```

Comment:
All the errors below occur.  As proposed we can enable and fix them one by one in step 3.